src/Controls/src/Core/BindingExpression.cs

```diff
@@ -668,6 +672,8 @@ class BindingExpressionPart
 			readonly PropertyChangedEventHandler _changeHandler;
 			WeakPropertyChangedProxy _listener;
 
+			~BindingExpressionPart() => _listener?.Unsubscribe(finalizer: true);
```

Comment:
This is the core fix, that allows `WeakPropertyChangedProxy` to be freed appropriately.